src/Native/Runtime/unix/UnixNativeCodeManager.cpp

```diff
@@ -145,9 +145,8 @@ bool UnixNativeCodeManager::UnwindStackFrame(MethodInfo *    pMethodInfo,
 
         GcInfoDecoder decoder(GCInfoToken(p), DECODE_REVERSE_PINVOKE_VAR);
 
-        // @TODO: CORERT: Encode reverse PInvoke frame slot in GCInfo: https://github.com/dotnet/corert/issues/2115
-        // INT32 slot = decoder.GetReversePInvokeFrameStackSlot();
-        // assert(slot != NO_REVERSE_PINVOKE_FRAME);
+        INT32 slot = decoder.GetReversePInvokeFrameStackSlot();
+        assert(slot != NO_REVERSE_PINVOKE_FRAME);
 
         *ppPreviousTransitionFrame = (PTR_VOID)-1;
```

Comment:
Could you also set `*ppPreviousTransitionFrame` to the address of the slot?

If I understand the JIT logic for the slot offset, it should call `GetFramePointer` and use that as a base if there is one; otherwise use GetSP() as the base.

Once you do that, could you also verify under debugger that it computes the right address for both frame pointer and non-framepointer cases?